File: backend/internal/model/transaction.go
package model

import "time"

type Transaction struct {
	ID         int64     `json:"id"`
	Amount     float64   `json:"amount"`
	CardID     int64     `json:"card_id"`
	TerminalID int64     `json:"terminal_id"`
	CreatedAt  time.Time `json:"created_at"`
}

type CreateTransactionRequest struct {
	Amount     float64 `json:"amount" binding:"required"`
	CardID     int64   `json:"card_id" binding:"required"`
	TerminalID int64   `json:"terminal_id" binding:"required"`
}

type AuthTransactionRequest struct {
	CardNumber string  `json:"card_number" binding:"required"`
	Amount     float64 `json:"amount" binding:"required"`
	TerminalID int64   `json:"terminal_id" binding:"required"`
}

type AuthTransactionResponse struct {
	Authorized bool   `json:"authorized"`
	Message    string `json:"message"`
}

File: frontend/src/commonMain/kotlin/com/transportpay/data/models/Transaction.kt
package com.transportpay.data.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Transaction(
    val id: Int,
    val amount: Double,
    @SerialName("card_id") val cardId: Int,
    @SerialName("terminal_id") val terminalId: Int,
    @SerialName("created_at") val createdAt: String
)

@Serializable
data class CreateTransactionRequest(
    val amount: Double,
    @SerialName("card_id") val cardId: Int,
    @SerialName("terminal_id") val terminalId: Int
)

@Serializable
data class AuthTransactionRequest(
    @SerialName("card_number") val cardNumber: String,
    val amount: Double,
    @SerialName("terminal_id") val terminalId: Int
)

@Serializable
data class AuthTransactionResponse(
    val authorized: Boolean,
    val message: String
)
